packages/textfield/test/textfield.test.ts

```diff
@@ -429,6 +431,49 @@ describe('Textfield', () => {
             : null;
         expect(input).to.not.be.null;
     });
+    it('valid - boundary-type assertions and title', async () => {
+        const el = await litFixture<Textfield>(
+            html`
+                <sp-textfield
+                    placeholder="Enter your number"
+                    pattern="^[\\d]+$"
+                    value="123"
+                ></sp-textfield>
+            `
+        );
+        await elementUpdated(el);
+
+        expect(el).to.not.equal(undefined);
+        const input = el.shadowRoot
+            ? el.shadowRoot.querySelector('#valid')
+            : null;
```

Comment:
```suggestion
        const input = el.shadowRoot.querySelector('#valid');
```

We likely still have this pattern in a number of places, however the `shadowRoot` should be guaranteed by TS and no longer need this pattern.